hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java

```diff
@@ -207,9 +209,13 @@ private long copyToFile(Path targetPath, FileSystem targetFS,
 
     ErasureCodingPolicy ecPolicy = null;
     if (preserveEC && sourceStatus.isErasureCoded()
-        && sourceStatus instanceof HdfsFileStatus
-        && targetFS instanceof DistributedFileSystem) {
-      ecPolicy = ((HdfsFileStatus) sourceStatus).getErasureCodingPolicy();
+        && sourceFS instanceof WithErasureCoding
+        && targetFS instanceof WithErasureCoding) {
+      String ecPolicyName =
+          ((WithErasureCoding) sourceFS).getErasureCodingPolicyName(sourceStatus);
+      if (ecPolicyName != null) {
+        ecPolicy = SystemErasureCodingPolicies.getByName(ecPolicyName);
```

Comment:
is this needed now the policy is being passed across as a string? as it is requiring that HDFS files define which ECs are available and mandates that hadoop-hdfs is on the classpath.

Reply:
Done.